optimum/onnx/graph_transformations.py

```
@@ -79,3 +97,227 @@ def remove_duplicate_weights(model: ModelProto, inplace: bool = False) -> ModelP
+
+    model_simp, check = simplify(model)
+    if not check:
+        raise RuntimeError(f"Simplified ONNX model could not be validated. Please turn off ONNX simplifier.")
+    if print_info:
+        print_simplifying_info(model, model_simp)
+
+    return model_simp
+
+
+def _deduplicated_cross_model_initializers(models: List[ModelProto], suffix: str = None):
+
+    all_initializers = []
+    for model in models:
+        all_initializers += list(model.graph.initializer)
+
+    duplicates = _find_duplicate_initializers(all_initializers)
+    name_sharing_dict = _create_name_sharing_dict(duplicates, suffix=suffix)
+    for model in models:
+        _replace_input_names(model, name_sharing_dict)
+
+    deduplicated_initializers = []
+    deduplicated_name = set()
+
+    for initializer in all_initializers:
+        if name_sharing_dict[initializer.name] not in deduplicated_name:
+            deduplicated_name.add(name_sharing_dict[initializer.name])
+            initializer.name = name_sharing_dict[initializer.name]
+            deduplicated_initializers.append(initializer)
+
+    return deduplicated_initializers
+
+
+def merge_decoders(
+    decoder: ModelProto,
+    decoder_with_past: ModelProto,
+    graph_name: str = "merged",
+    producer_name: str = "optimum-onnx",
+    simplify_graph: bool = False,
+) -> ModelProto:
+    """
+    Fuses decoder ONNX model and decoder with past ONNX model into one ONNX model with if logic.
+
+    Args:
+        decoder (`onnx.ModelProto`): Decoder ONNX model.
+        decoder_with_past (`onnx.ModelProto`): Decoder with past ONNX model.
+        graph_name (`str`): Name of the parent graph(graph of the control flow node).
+        producer_name (`str`): Graph producer name.
+        simplify_graph (`bool`): Whether to use onnx-simplifier to simplify subgraphs.
+
+    Returns:
+        `~onnx.ModelProto`: The fused decoder ONNX model.
+    """
+    if simplify_graph:
```

can we raise an error if `not is_onnxsim_available()`?